### ISSUE TITLE
Windows issue moving from v0.42.0 to v0.43.0+

### ISSUE THREAD

**Issue by Maisha** [rollup/rollup#1483]
After upgrading to rollup v0.43.0+ i get information about gyp errors at `npm install`

`npm install` with this package.json do not give any gyp error output. 
```
{
  "license": "MIT",
  "name": "rolluptest",
  "version": "0.0.1",
  "description": "test",
  "author": {
    "name": "Michael Lund Rasmussen",
    "email": "some@email.com"
  },
  "main": "main.js",
  "repository": {
    "type": "git",
    "url": "ssh://somerepo.git"
  },
  "devDependencies": {
    "rollup": "^0.42.0"
  }
}
```
gives: 
```
added 3 packages in 1.547s
```
And running `npm install` again gives:
```
up to date in 1.335s
```

Where:
`npm install` with this package.json: 
```
{
  "license": "MIT",
  "name": "rolluptest",
  "version": "0.0.1",
  "description": "test",
  "author": {
    "name": "Michael Lund Rasmussen",
    "email": "some@email.com"
  },
  "main": "main.js",
  "repository": {
    "type": "git",
    "url": "ssh://somerepo.git"
  },
  "devDependencies": {
    "rollup": "^0.45.0"
  }
}
```
Seems to install but, gives gyp Err
```
> weak@1.0.1 install C:\code\rolluptest\node_modules\weak
> node-gyp rebuild

C:\code\rolluptest\node_modules\weak>if not defined npm_config_node_gyp (node "C:\Users\mlr\AppData\Roaming\npm\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "" rebuild )
gyp ERR! configure error
gyp ERR! stack Error: Can't find Python executable "python", you can set the PYTHON env variable.
gyp ERR! stack     at PythonFinder.failNoPython (C:\Users\mlr\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\configure.js:483:19)
gyp ERR! stack     at PythonFinder.<anonymous> (C:\Users\mlr\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\configure.js:508:16)
gyp ERR! stack     at C:\Users\mlr\AppData\Roaming\npm\node_modules\npm\node_modules\graceful-fs\polyfills.js:284:29
gyp ERR! stack     at FSReqWrap.oncomplete (fs.js:114:15)
gyp ERR! System Windows_NT 10.0.14393
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\mlr\\AppData\\Roaming\\npm\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\xampp\htdocs\dtm\rolluptest\node_modules\weak
gyp ERR! node -v v7.7.2
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: weak@1.0.1 (node_modules\weak):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: weak@1.0.1 install: `node-gyp rebuild`
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Exit status 1

added 6 packages in 3.163s
```

and running `npm install` again do not give the expected: 
```
up to date in [xxxx]s'
```

but gives
```
> weak@1.0.1 install C:\xampp\htdocs\dtm\rolluptest\node_modules\weak
> node-gyp rebuild


C:\code\rolluptest\node_modules\weak>if not defined npm_config_node_gyp (node "C:\Users\mlr\AppData\Roaming\npm\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "" rebuild )
gyp ERR! configure error
gyp ERR! stack Error: Can't find Python executable "python", you can set the PYTHON env variable.
gyp ERR! stack     at PythonFinder.failNoPython (C:\Users\mlr\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\configure.js:483:19)
gyp ERR! stack     at PythonFinder.<anonymous> (C:\Users\mlr\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\configure.js:508:16)
gyp ERR! stack     at C:\Users\mlr\AppData\Roaming\npm\node_modules\npm\node_modules\graceful-fs\polyfills.js:284:29
gyp ERR! stack     at FSReqWrap.oncomplete (fs.js:114:15)
gyp ERR! System Windows_NT 10.0.14393
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\mlr\\AppData\\Roaming\\npm\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\code\rolluptest\node_modules\weak
gyp ERR! node -v v7.7.2
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: weak@1.0.1 (node_modules\weak):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: weak@1.0.1 install: `node-gyp rebuild`
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Exit status 1

added 3 packages in 2.703s
```

**Comment by Rich-Harris** [comment]
Can you try using 0.45? `weak` is now an `optionalDependency` rather than a `devDependency`, precisely because of this issue (my bad — I didn't realise it doesn't work on Windows).

**Comment by dmitrage** [comment]
`weak` should better be "optional**Dev**Dependency", but AFAIK there is no such thing in npm :(

Maybe it's better to exclude it from package.json and change `test:leak` script to something like `npm i --no-save --silent weak@1.0.1 || node --expose-gc test/leak/test.js`?

**Comment by Rich-Harris** [comment]
Have released 0.45.1 which makes that change — we should be all good now. Thanks for the report
